taskcluster/ci/android-build/kind.yml

```diff
@@ -30,15 +28,19 @@ jobs:
     worker:
       docker-image: { in-tree: linux }
       max-run-time: 1800
-      script: |
-        rsync -a /builds/worker/fetches/libs/ /builds/worker/checkouts/src/libs/
-        echo "rust.targets=linux-x86-64,x86_64\n" > local.properties
-        ./gradlew --no-daemon clean
-        ./gradlew --no-daemon assembleDebug
-        ./gradlew --no-daemon testDebug
     run:
-      using: run-task
-      cwd: "{checkout}"
+      pre-commands:
+        - [./taskcluster/scripts/rustup-setup.sh]
+        - [./taskcluster/scripts/cross-compile-setup.sh]
+      commands:
+        - [rsync, '-a', /builds/worker/fetches/libs/, /builds/worker/checkouts/src/libs/]
+        - [echo, 'rust.targets=linux-x86-64,x86_64\n', '>', local.properties]
+        - [./gradlew, '--no-daemon', 'clean']
+        - [./gradlew, '--no-daemon', 'assembleDebug']
+        - [./gradlew, '--no-daemon', 'testDebug']
+      using: run-commands
+      use-caches: true
```

Comment:
We don't have it set because the caching story with git and `run-task` is not nearly as robust as we have via `hg robustcheckout`. I recall there were issues with the wrong project being checked out. I don't recall if we ever dug into the details. It is much less of an issue given the relative sizes of the repositories that we are cloning, in any case.

Reply:
TIL, good to know, thank you!